### ISSUE TITLE
generate plugin config on init

### ISSUE THREAD

**Issue by godcong** [ipfs/go-ipfs#7675]
From the current version, it seems that the ipfs plugin does not support the default configuration plugin config.
Although plugin.Environment defines `Config` interface{}, But it doesn’t seem to be used well.

A plugin may have its own private configuration. I think it would be better if the default configuration of the plugin can be read during init.
This may only require a hook to call an interface implemented by the plugin.
Then, you can use this configuration directly in subsequent uses without any manual operation.

For Example: 
if you have a plugin named "plugin-foo", when you run 'ipfs init'.
the init calls some hook function,then append the default config into Plugins.
```
{
  // ...
    "Plugins": {
      // plugin named "plugin-foo"
      "plugin-foo": {
        "Config": { /* arbitrary json */ }  //<- append into Config on init
      },
    }
}
```

How about this idea?

**Comment by aschmahmann** [comment]
I'll try and take a look at this next week. @Stebalien any thoughts on this? I haven't looked into Plugin configs so no opinions on it just yet.

**Comment by Stebalien** [comment]
I don't think this is the right approach. In general, the default config should be empty as it makes it easier to upgrade/change defaults.

**Comment by godcong** [comment]
@Stebalien 
I see it as an extended function developed at the user level. This part of the function may use `Sql`, `Redis` or some other places where parameters must be configured.

Just like the default port of `IPFS`, once it is occupied by others, you have to use other ports, which requires modifying the Config file.
If there is no default configuration, it is possible to configure the wrong format.
Moreover, the configuration is best generated and updated through Plugin's own way.
